src/napari_phasors/_reader.py

```diff
@@ -206,27 +206,38 @@ def raw_file_reader(
         and 'frequency' in raw_data.attrs.keys()
     ):
         settings['frequency'] = raw_data.attrs['frequency']
+
     layers = []
     iter_axis = iter_index_mapping[file_extension]
-    if iter_axis is None:
+
+    if iter_axis is None or iter_axis not in raw_data.dims:
+        # Handle files without iteration axis or when keepdims=False squeezed it out
         if file_extension == ".tif":
-            mean_intensity_image, G_image, S_image = phasor_from_signal(
-                raw_data, axis=0, harmonic=harmonics
-            )
+            axis = 0
+        elif iter_axis is None:
+            # Hyperspectral files - find "C" dimension
+            axis = raw_data.dims.index("C")
         else:
-            # Calculate phasor over channels if file is of hyperspectral type
-            mean_intensity_image, G_image, S_image = phasor_from_signal(
-                raw_data, axis=raw_data.dims.index("C"), harmonic=harmonics
-            )
+            # FLIM files without "C" dimension (single channel)
+            axis = raw_data.dims.index("H")
+
+        mean_intensity_image, G_image, S_image = phasor_from_signal(
+            raw_data, axis=axis, harmonic=harmonics
+        )
         labels_layer = make_phasors_labels_layer(
             mean_intensity_image,
             G_image,
             S_image,
             name=filename,
             harmonics=harmonics,
         )
+        channel_suffix = (
+            " Intensity Image"
+            if iter_axis is None
+            else " Intensity Image: Channel 0"
+        )
         add_kwargs = {
-            "name": f"{filename} Intensity Image",
+            "name": f"{filename}{channel_suffix}",
```

Comment:
It might be enough to just specify the `channel_axis` argument in the kwargs of the napari LayerData tuple. For example:
`"channel_axis": 0,` below this line in cases where we have the unitary extra channel dimension in the first axis.